codalab/rest/account.py

```diff
@@ -114,28 +86,42 @@ def do_signup():
         errors.append("User with this username or email already exists.")
 
     if errors:
-        return template('signup', errors=errors)
+        return redirect_with_query(error_uri, {
+            "error": " ".join(errors),
+            "next": success_uri,
+            "email": email,
+            "username": username,
+        })
 
     # Create unverified user
     _, verification_key = local.model.add_user(username, email, password)
 
-    return send_verification_key(username, email, verification_key)
+    # Send key
+    send_verification_key(username, email, verification_key)
+
+    # Redirect to success page
+    return redirect_with_query(success_uri, {
+        "email": email
+    })
 
 
 @get('/account/verify/<key>', skip=UserVerifiedPlugin)
 def do_verify(key):
     if local.model.verify_user(key):
-        return redirect(default_app().get_url('success', message="Account verified!"))
+        return redirect('/account/verify/success')
     else:
-        return "Invalid or expired verification key."
+        return redirect('/account/verify/error')
 
 
 @get('/account/resend', name='resend_key', skip=UserVerifiedPlugin)
```

Comment:
Is it actually used?
On Mar 14, 2016 8:57 PM, "Stephen Koo" <notifications@github.com> wrote:

> In codalab/rest/account.py
> <https://github.com/codalab/codalab-cli/pull/396#discussion_r56112542>:
>
> >      else:
> > -        return "Invalid or expired verification key."
> > +        return redirect('/account/verify/error')
> >
> >
> >  @get('/account/resend', name='resend_key', skip=UserVerifiedPlugin)
>
> This is for if someone's email client screws up the link, or the link is
> miscopied. It might be possible :P
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub:
> https://github.com/codalab/codalab-cli/pull/396/files#r56112542
>